charts/kube-prometheus-stack/Chart.yaml

```diff
@@ -23,8 +23,8 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 55.11.0
-appVersion: v0.70.0
+version: 56.0.0
+appVersion: v0.71.0
```

Comment:
It looks like 0.71.0 version of Prometheus Operator starts config-reloader side-car container of Alertmanager and Prometheus with TLS enabled if TLS is enabled for the main container (e.g., using .alertmanager.alertmanagerSpec.web.tlsConfig and .prometheus.prometheusSpec.web.tlsConfig value of kube-promethues-stack Helm chart):

```text
$ kubectl logs alertmanager-prometheus-kube-prometheus-alertmanager-0 -n monitoring -c config-reloader
d0)"
level=info ts=2024-01-22T04:21:35.063518686Z caller=main.go:138 build_context="(go=go1.21.5, platform=linux/amd64, user=Action-Run-ID-7500027263, date=20240112-09:04:19, tags=unknown)"
level=info ts=2024-01-22T04:21:35.063805761Z caller=reloader.go:246 msg="reloading via HTTP"
level=info ts=2024-01-22T04:21:35.066813295Z caller=main.go:193 msg="Starting web server for metrics" listen=:8080
level=info ts=2024-01-22T04:21:35.071097943Z caller=tls_config.go:313 msg="Listening on" address=[::]:8080
level=info ts=2024-01-22T04:21:35.071741226Z caller=tls_config.go:349 msg="TLS is enabled." http2=true address=[::]:8080
level=info ts=2024-01-22T04:21:35.074918397Z caller=reloader.go:424 msg="Reload triggered" cfg_in=/etc/alertmanager/config/alertmanager.yaml.gz cfg_out=/etc/alertmanager/config_out/alertmanager.env.yaml watched_dirs=/etc/alertmanager/config
level=info ts=2024-01-22T04:21:35.074990138Z caller=reloader.go:282 msg="started watching config file and directories for changes" cfg=/etc/alertmanager/config/alertmanager.yaml.gz out=/etc/alertmanager/config_out/alertmanager.env.yaml dirs=/etc/alertmanager/config
```

But the ServiceMonitor which monitors config-reloader side-car container of Alertmanager and Prometheus uses hard-coded `http` scheme (https://github.com/prometheus-community/helm-charts/blob/kube-prometheus-stack-56.0.1/charts/kube-prometheus-stack/templates/alertmanager/servicemonitor.yaml#L55 and https://github.com/prometheus-community/helm-charts/blob/kube-prometheus-stack-56.0.1/charts/kube-prometheus-stack/templates/prometheus/servicemonitor.yaml#L48) which respective target scraping failing and Prometheus alerts triggered.

Reply:
@DrFaust92 could you please take a look here?

Reply:
@jkroepke @mabrarov friendly ping for the above

Reply:
Hi @DrFaust92,

I agree to have 1 (it will unblock my team) and then 2 (it will make chart more correct logically, but it can introduce a change which can have potential negative impact, so it requires major release and special note in release notes).

Thank you.

Reply:
I think this could be sufficient https://github.com/prometheus-community/helm-charts/pull/4230